buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1881,35 +1892,23 @@ void transferTo(ByteBuf dst) {
         }
 
         ByteBuf slice() {
-            return slice != null ? slice : (slice = buf.slice(idx(offset), length()));
+            return slice != null ? slice : (slice = srcBuf.slice(srcIdx(offset), length()));
```

Comment:
please make this more readable... Sure its compact but its really hard to read.

Reply:
done